administrator/components/com_associations/models/fields/associatedcomponent.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_associations
+ *
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+defined('JPATH_BASE') or die;
+
+JFormHelper::loadFieldClass('groupedlist');
+/**
+ * A drop down containing all components that implement associations
+ *
+ * @package     Joomla.Administrator
+ * @subpackage  com_associations
+ * @since       __DEPLOY_VERSION__
+ */
+class JFormFieldAssociatedComponent extends JFormFieldGroupedList
+{
+	/**
+	 * The form field type.
+	 *
+	 * @var    string
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $type = 'AssociatedComponent';
+	
+	/**
+	 * Method to get the field input markup.
+	 *
+	 * @return  string	The field input markup.
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function getGroups()
+	{
+		jimport('joomla.filesystem.folder');
+		jimport('joomla.filesystem.file');
+
+		$options = array();
+
+		$componentsDirectory         = JPATH_ADMINISTRATOR . '/components';
+		$frontendComponentsDirectory = JPATH_SITE . '/components';
+		$backendComponents           = glob($componentsDirectory . '/*', GLOB_NOSORT | GLOB_ONLYDIR);
+		$frontendComponents          = glob($frontendComponentsDirectory . '/*', GLOB_NOSORT | GLOB_ONLYDIR);
+		
+		// Keeping only directory name
+		for ($i = 0; $i < count($backendComponents); $i++)
+		{ 
+			$backendComponents[$i] = basename($backendComponents[$i]);
```

Comment:
do we really need an exclusive for cycle just to do this when we have a similiar for in the lines below?